cpp/tests/rolling/grouped_rolling_range_test.cpp

```diff
@@ -287,3 +288,321 @@ TYPED_TEST(GroupedRollingRangeOrderByDecimalTypedTest, UnboundedRanges)
+  cudf::range_window_bounds const unbounded_preceding =
+    cudf::range_window_bounds::unbounded(string_type);
+  cudf::range_window_bounds const unbounded_following =
+    cudf::range_window_bounds::unbounded(string_type);
+  cudf::range_window_bounds const current_row = cudf::range_window_bounds::current_row(string_type);
+
+  [[nodiscard]] auto get_count_over_partitioned_window(
+    cudf::column_view const& order_by,
+    cudf::order const& order,
+    cudf::range_window_bounds const& preceding,
+    cudf::range_window_bounds const& following) const
+  {
+    return cudf::grouped_range_rolling_window(
+      cudf::table_view{{*grouping_keys}},
+      order_by,
+      order,
+      *agg_values,
+      preceding,
+      following,
+      min_periods,
+      *cudf::make_count_aggregation<cudf::rolling_aggregation>());
+  }
+
+  [[nodiscard]] auto get_count_over_unpartitioned_window(
+    cudf::column_view const& order_by,
+    cudf::order const& order,
+    cudf::range_window_bounds const& preceding,
+    cudf::range_window_bounds const& following) const
+  {
+    return cudf::grouped_range_rolling_window(
+      cudf::table_view{std::vector<cudf::column_view>{}},
+      order_by,
+      order,
+      *agg_values,
+      preceding,
+      following,
+      min_periods,
+      *cudf::make_count_aggregation<cudf::rolling_aggregation>());
+  }
+};
+
+TEST_F(GroupedRollingRangeOrderByStringTest, Ascending_Partitioned_NoNulls)
+{
+  // clang-format off
+  auto const orderby =
+      strings_column{
+          "A", "A", "A", "B", "B", "B", // Group 0.
+          "C", "C", "C", "C",           // Group 1.
+          "D", "D", "E", "E"            // Group 2.
+      }.release();
+  // clang-format on
+
+  // Partitioned cases.
+  CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(
+    *get_count_over_partitioned_window(
+      *orderby, cudf::order::ASCENDING, unbounded_preceding, current_row),
+    ints_column{3, 3, 3, 6, 6, 6, 4, 4, 4, 4, 2, 2, 4, 4});
+
+  CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(
+    *get_count_over_partitioned_window(
+      *orderby, cudf::order::ASCENDING, current_row, unbounded_following),
+    ints_column{6, 6, 6, 3, 3, 3, 4, 4, 4, 4, 4, 4, 2, 2});
+
+  CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(
+    *get_count_over_partitioned_window(
+      *orderby, cudf::order::ASCENDING, unbounded_preceding, unbounded_following),
+    ints_column{6, 6, 6, 6, 6, 6, 4, 4, 4, 4, 4, 4, 4, 4});
+
+  CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(
+    *get_count_over_partitioned_window(*orderby, cudf::order::ASCENDING, current_row, current_row),
+    ints_column{3, 3, 3, 3, 3, 3, 4, 4, 4, 4, 2, 2, 2, 2});
+}
+
+TEST_F(GroupedRollingRangeOrderByStringTest, Ascending_NoParts_NoNulls)
+{
+  // clang-format off
+  auto const orderby =
+      strings_column{
+          "A", "A", "A", "B", "B", "B",
+          "C", "C", "C", "C",
+          "D", "D", "E", "E"
+      }.release();
+  // clang-format on
+
+  // Un-partitioned cases.
+  CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(
```

Comment:
Nit: Please try to use the `_EQUAL` assert version if possible, so we can check the output exactly.

Reply:
Fixed.